xc7/tests/soc/litex/mini_ddr/scripts/min_ddr_arty.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+# This file is Copyright (c) 2015-2019 Florent Kermarrec <florent@enjoy-digital.fr>
+# License: BSD
+
+import argparse
+
+from migen import Module, ClockDomain, Signal, Instance
+
+from litex.boards.platforms import arty
+from litex.build.xilinx.vivado import vivado_build_args, vivado_build_argdict
+
+from litex.soc.cores.clock import S7PLL, S7IDELAYCTRL
+from litex.soc.integration.soc_sdram import SoCSDRAM, soc_sdram_args, soc_sdram_argdict
+from litex.soc.integration.builder import Builder, builder_args, builder_argdict
+
+from litedram.modules import MT41K128M16
+from litedram.phy import s7ddrphy
+
+# CRG ----------------------------------------------------------------------------------------------
+
+
+class _CRG(Module):
+    def __init__(self, platform, sys_clk_freq):
+        self.clock_domains.cd_sys = ClockDomain()
+        self.clock_domains.cd_sys4x = ClockDomain(reset_less=True)
+        self.clock_domains.cd_sys4x_dqs = ClockDomain(reset_less=True)
+        self.clock_domains.cd_clk200 = ClockDomain()
+        self.clock_domains.cd_eth = ClockDomain()
+
+        # # #
+        pll_clkin = Signal()
+        self.specials += Instance(
+            "BUFG", i_I=platform.request("clk100"), o_O=pll_clkin
+        )
+        self.submodules.pll = pll = S7PLL(speedgrade=-1)
+        self.comb += pll.reset.eq(~platform.request("cpu_reset"))
+        pll.register_clkin(pll_clkin, 100e6)
+        pll.create_clkout(self.cd_sys, sys_clk_freq)
+        pll.create_clkout(self.cd_sys4x, 4 * sys_clk_freq)
+        pll.create_clkout(self.cd_sys4x_dqs, 4 * sys_clk_freq, phase=90)
+        pll.create_clkout(self.cd_clk200, 200e6)
+        self.submodules.idelayctrl = S7IDELAYCTRL(self.cd_clk200)
```

Comment:
An additional thing would be to add the output to the idelayctrl (RDY signal), as currently VPR still removes it.

It could simply be directed to a led